code/game/area/areas.dm

```diff
@@ -475,29 +475,12 @@ GLOBAL_LIST_EMPTY(teleportlocs)
 	for(var/obj/machinery/light/L in src)
 		L.update(TRUE, TRUE, TRUE)
 
-/**
-  * Update the icon state of the area
-  *
-  * Im not sure what the heck this does, somethign to do with weather being able to set icon
-  * states on areas?? where the heck would that even display?
-  */
-/area/update_icon_state()
-	var/weather_icon
-	for(var/V in SSweather.processing)
-		var/datum/weather/W = V
-		if(W.stage != END_STAGE && (src in W.impacted_areas))
-			W.update_areas()
-			weather_icon = TRUE
-	if(!weather_icon)
-		icon_state = null
-	return ..()
```

Comment:
This is called every time APC power state is changed, and that's the reason why areas have additional weather appearance.
This also does wrong with the old weather code too. It should be definitely gone.